sdk/core/azure-core/src/http/transport_policy.cpp

```diff
@@ -14,8 +22,10 @@ std::shared_ptr<HttpTransport> Azure::Core::Http::Details::GetTransportAdapter()
   return ::AzureSdkGetCustomHttpTransport();
 #elif defined(BUILD_TRANSPORT_WINHTTP_ADAPTER)
   return std::make_shared<Azure::Core::Http::WinHttpTransport>();
-#else
+#elif defined(BUILD_CURL_HTTP_TRANSPORT_ADAPTER)
   return std::make_shared<Azure::Core::Http::CurlTransport>();
+#else
+  return std::shared_ptr<HttpTransport>();
```

Comment:
Does this change adhere to the behavior we want? Why do we have this `else` case and what does returning a pointer to an unimplemented `HttpTransport` base class mean?

If the user doesn't explicitly pick anything (i.e. use the SDK right out of the box), we want curl to be used by default on linux/posix.

Reply:
We have to build in "no transport" configuration, at least for VcPkg. When you tell it to build "azure-core-cpp[core]", it should build an implementation without any transport. Unless we want to make curl a permanent dependency for all configurations, which we don't want to do. But if we do, we ned to remove `curl` vcpkg option, if we always rely on Curl.